src/main/java/eu/europa/ted/efx/xpath/XPathScriptGenerator.java

```diff
@@ -471,6 +471,32 @@ public DurationExpression composeSubtraction(DurationExpression left, DurationEx
     return new DurationExpression("(" + left.script + " - " + right.script + ")");
   }
 
+
+  @Override
+  public <T extends Expression, L extends ListExpression<T>> L composeDistinctValuesFunction(
+      L listOne, L listTwo, Class<L> listType) {
+        return Expression.instantiate("distinct-values(" + listOne.script + ", " + listTwo.script + ")", listType);
+  }
+
+  @Override
+  public <T extends Expression, L extends ListExpression<T>> L composeUnionFunction(L listOne,
+      L listTwo, Class<L> listType) {
+        return Expression.instantiate("distinct-values(" + listOne.script + ", " + listTwo.script + ")", listType);
```

Comment:
Parentheses also missing here.

Reply:
Parentheses still missing here.
Currently this gives: distinct-values(listOne, listTwo)
It should be: distinct-values((listOne, listTwo))

Reply:
That does not make much sense. What is the semantic of the extra parenthesis?

Reply:
It turns out you are right. I looked it up and fixed it. Thanks

Reply:
It's to make a single sequence from the 2 sequences. Otherwise you're calling distinct-values with 2 parameters that are  sequences, which is not allowed.
That's how value-union is implemented here:
http://www.xqueryfunctions.com/xq/functx_value-union.html